src/task/models/tileMergeTaskManager.ts

```diff
@@ -243,14 +270,41 @@ export class TileMergeTaskManager {
 
     span.setAttributes({ partsZoomLevelMatch, maxZoom, ppCollectionLength: ppCollection.features.length });
 
-    for (let zoom = maxZoom; zoom >= 0; zoom--) {
+    // Store original resume state (NEVER changes during loop)
+    const resumeFromZoom = initTask.parameters.taskIndex?.zoomLevel ?? maxZoom;
+    const resumeFromTaskIndex = initTask.parameters.taskIndex?.lastInsertedTaskIndex ?? 0;
+
+    logger.info({
```

Comment:
this log will appear even when state isn't resume should be wrapped by an if condition depends on "initTask" `taskIndex` parameter existence

Reply:
change the log so it will indicate for what task/zoom point its started.